### ISSUE TITLE
SuperTiled2Unity version: 2.1.1, Unity version: 2022.3.17f1c1 Unknown error encountered. Please report as bug. Stack track is in the console output. Failed to add object of type `SuperTile`. Check that the definition is in a file of the same name and that it compiles properly.

### ISSUE THREAD

**Issue by 751500365** [Seanba/SuperTiled2Unity#268]
SuperTiled2Unity version: 2.1.1, Unity version: 2022.3.17f1c1
Unknown error encountered. Please report as bug. Stack track is in the console output.
Failed to add object of type `SuperTile`. Check that the definition is in a file of the same name and that it compiles properly.
![image](https://github.com/Seanba/SuperTiled2Unity/assets/12488622/797e47a4-bf92-4d65-9cc8-40d1ae181675)
![screenshot-20240430-104452](https://github.com/Seanba/SuperTiled2Unity/assets/12488622/4941746b-c345-4e3b-a080-3a968df85811)
![screenshot-20240509-181725](https://github.com/Seanba/SuperTiled2Unity/assets/12488622/9bceb942-bc25-42d0-a6a1-65e154210c87)

**Comment by Seanba** [comment]
Hi there, @751500365. Are there any compile errors in the Unity Console window? It looks like something in the ST2U package didn't build correctly. Is it possible for me to get a copy of your project to debug further if needed?

**Comment by Rich-DOGLtd** [comment]
Hi - I am getting this error on every Tiled file (Levels, Tilesets), so currently can't load any maps.

![image](https://github.com/user-attachments/assets/f0609807-7903-4235-994a-5b9ba810034a)

**Comment by Rich-DOGLtd** [comment]
There are no compile errors.

**Comment by Rich-DOGLtd** [comment]
> Hi there, @751500365. Are there any compile errors in the Unity Console window? It looks like something in the ST2U package didn't build correctly. Is it possible for me to get a copy of your project to debug further if needed?

I am happy to send you the project source if you need it.

**Comment by Seanba** [comment]
Hi there, @Rich-DOGLtd. Yes, if you can send me the project I'd be happy to investigate it further. I haven't seen this issue personally myself so I'm interested in finding out what's going on.

**Comment by Rich-DOGLtd** [comment]
Fantastic - thanks Sean. Will send link via email.

**Comment by Rich-DOGLtd** [comment]
I've just sent over the link - it is a zip archive on my Google Drive.

**Comment by Seanba** [comment]
Thanks, @Rich-DOGLtd. I was able to reproduce the bug in your project but I don't yet know the cause.

Can you try to force reimport the `Super Tiled2Unity` package under `Packages/Super Tiled2Unity`? And then take one of your tilesets and force reimport that as well?

That got things working for me though I still want to investigate more on what's going on here.

![image](https://github.com/user-attachments/assets/be9a78a7-276a-4096-b5ef-fd76c5f80353)

**Comment by Seanba** [comment]
@Rich-DOGLtd I think closing Unity, deleting your `Library` folder, and reopening Unity can fix things up for you too.

For some reason your project import is in a confused state and forcing everything to be reimported into your library can fix that up.

**Comment by Rich-DOGLtd** [comment]
Ah - ok, thanks Sean - I will give that a try. Thanks for taking a look :)

Richard

**Comment by Seanba** [comment]
@Rich-DOGLtd after further testing I've confirmed that deleting your `Library` is the right move to fix most of your issues.

**However**, your `Level01.tmx` Tiled map is using multiple embedding tilesets and, after all this time, that's something ST2U hasn't accounted for. You'll still see errors like this for that map until I provide a fix which should be ready sometime tomorrow.

Thanks for helping me track this down.

**Comment by Seanba** [comment]
@Rich-DOGLtd I've posted [version v2.2.3 of Super Tiled2Unity](https://seanba.itch.io/supertiled2unity/devlog/846462/super-tiled2unity-223). After deleting your `Library` folder trying replacing your install of ST2U and re-open Unity.

Note that Unity 6 now requires this naming convention on ST2U as an embedded package: `Packages/com.seanba.super-tiled2unity/` 
Because of this I've renamed the zip file to `com.seanba.super-tiled2unity.zip`.
